recipes-bsp/u-boot/u-boot-distro-boot/uEnv.txt.in

```diff
@@ -108,6 +112,116 @@ bootcmd_boot=if test ${bootscript_debug} != 1; then \
                  fi; \
              fi || true
 
-bootcmd_run=run board_fixups && run check_rollback_needed && run set_bootargs && run set_fdt_path && \
+prog_fuses=if test -n "${fuse_prog_list}"; then \
+               for fuse_val in "${fuse_prog_list}"; do \
+                   if test "${fuse_fail}" != "1"; then \
+                       if test "${fuse_dry_run}" = "1"; then \
+                           echo "DRY RUN: fuse prog ${fuse_bank} ${fuse_word} ${fuse_val}"; \
+                       else \
+                           fuse prog -y ${fuse_bank} ${fuse_word} ${fuse_val}; \
+                       fi; \
+                       if test $? != "0"; then \
+                           env set fuse_fail 1; \
+                       fi; \
+                       if test "${fuse_dry_run}" = "1"; then \
+                           echo "DRY RUN: fuse cmp"; \
+                       else \
+                           fuse cmp ${fuse_bank} ${fuse_word} ${fuse_val}; \
+                       fi; \
+                       if test $? != "0"; then \
+                           env set fuse_fail 1; \
+                       fi; \
+                   fi ; \
+                   setexpr fuse_word $fuse_word + 1; \
+                   if test "${soc}" = "imx8m" || test "${soc}" = "imx7d"; then \
+                       if test "${fuse_word}" = "4"; then \
+                           setexpr fuse_bank $fuse_bank + 1; \
+                           setexpr fuse_word 0; \
+                       fi; \
+                   fi; \
+                   if test "${fuse_word}" = "72a"; then \
+                       env set fuse_word 730; \
+                   elif test "${fuse_word}" = "73a"; then \
+                       env set fuse_word 740; \
+                   fi; \
+               done; \
+               @@STATUS_CMD@@; \
+               if test $? != "0"; then \
+                   env set fuse_fail 1; \
+               fi; \
+               if test "${fuse_prog_close}" = "1" && test "${fuse_fail}" != "1"; then \
+                   if test "${fuse_dry_run}" = "1"; then \
+                       echo "DRY RUN: Closing is set"; \
+                   else \
+                       @@CLOSE_CMD@@; \
+                   fi; \
+                   if test $? != "0"; then \
+                       env set fuse_fail 1; \
+                   fi; \
+               fi; \
+               env set fuse_bank @@FUSE_BANK@@; \
+               env set fuse_word @@FUSE_WORD@@; \
+               env set save_vars 1; \
```

Comment:
Shouldn't we have some logic to stop trying to program the fuses? The same logic could also prevent the wearing of the flash area used by the U-Boot environment.

A suggestion would be: instead of a `fuse_fail` we could have a `fuse_status` which could be set to something like "pending", "failure" or "success"; and we'd try to program the fuses only if the work was still pending.

What do you think?

Reply:
What do you mean by "stop trying to program the fuses"?

Do you mean in the event that `fuse_prog_list` doesn't get unset? In which case `prog_fuses` would be run on every boot? In my mind I was thinking the eventual action handler would unset `fuse_prog_list` after a success/failure attempt.

Or I guess you're thinking about production programming where there's no action handler to unset `fuse_prog_list`? In which case I guess this would make sense. Though we'd need to remember to also set `fuse_status` from the action handler as well.

Reply:
Adjusted in V2. Using `fuse_status` with pending/success/failed states.